.github/workflows/ci.yml

```diff
@@ -0,0 +1,245 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
+      - 'design/**'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - 'docs/**'
+      - 'design/**'
+
+env:
+  # Common versions
+  GO_VERSION: '1.14'
+  GOLANGCI_VERSION: 'v1.31'
+  DOCKER_BUILDX_VERSION: 'v0.4.2'
+
+  # Common users. We can't run a step 'if secrets.AWS_USR != ""' but we can run
+  # a step 'if env.AWS_USR' != ""', so we copy these to succinctly test whether
+  # credentials have been provided before trying to run steps that need them.
+  DOCKER_USR: ${{ secrets.DOCKER_USR }}
+  AWS_USR: ${{ secrets.AWS_USR }}
+
+jobs:
+
+  lint:
+    runs-on: ubuntu-latest
+
```

Comment:
I usually add this line right after `runs-on` in all the jobs too, that gives us a native way (without code change) to skip ci, for whatever reason.

```yaml
if: "!contains(github.event.head_commit.message, '[ci skip]')"
```

Note that `[ci skip]` is by convention and can be changed to any arbitrary text and can be put anywhere in the commit message (title or body).

Reply:
Our current Jenkins pipeline uses a similar convention, but in practice we mostly only use it to skip CI when we know we're making changes that don't need it to run (e.g. editing markdown). This pipeline excludes some files (markdown, images) via `paths-ignore` so I'm hoping we can get away without needing to explicitly skip CI now.

Reply:
I cannot mention a solid use-case of this at the moment in the scope of crossplane codebases, specially when in a PR, but once in a while I had ended up in a situation that I needed to skip a pipeline (mostly publishing of some sort) on the master branch in which this became really handy.